07a_system_activity_plots.R

```diff
@@ -206,81 +206,149 @@ renderSystemPlot <- function(id) {
-  scale_x_continuous(labels = str_wrap(df$Time %>% unique(), width = 10), # x axis labels
-                   breaks = df$group.id %>% unique()) +
-  theme_void() + # totally clear all theme elements
-  theme(# add back in what theme elements we want
-    text = element_text(size = 16, colour = "#4e4d47"),
-    axis.text.x = element_text(size = 16),
-    axis.ticks.x = element_line(),
-    axis.line.x = element_line(colour = "#4e4d47", linewidth = 0.5),
-    # axis.ticks.length.x = unit(.15, "cm"),
-    plot.margin = unit(c(1, 1, 1, 1), "lines"),
-    legend.text = element_text(size = 16),
-    legend.title = element_blank(),
-    legend.position = "bottom",
-    legend.margin = margin(.5, 0, 0, 0, unit = "inch")
-  )
-  })
- # return(plotOutput(id, height = 400))
-} 
+      ggplot(df, aes(x = group.id, fill = PlotFillGroups)) +
+        geom_rect(
+          # the bars
+          aes(
+            xmin = group.id - 0.25,
+            # control bar gap width
+            xmax = group.id + 0.25,
+            ymin = ystart,
+            ymax = yend
+          ),
+          colour = "#4e4d47",
+          linewidth = .2,
+          alpha = 0.8
+        ) +
+        geom_segment(
+          # the connecting segments between bars
+          data = df %>%
+            filter(group.id == group.id) %>%
+            group_by(group.id) %>%
+            slice_tail() %>%
+            ungroup() %>%
+            select(group.id, yend),
+          aes(
+            x = group.id,
+            xend = if_else(group.id == last(group.id), last(group.id), group.id + 1),
+            y = yend,
+            yend = yend
+          ),
+          linewidth = .3,
+          colour = "gray25",
+          linetype = "dashed",
+          show.legend = FALSE,
+          inherit.aes = FALSE
+        ) +
+        ggrepel::geom_text_repel(
+          # the labels
+          aes(
+            x = group.id,
+            label = paste0(scales::comma(abs(values))),
+            y = rowSums(cbind(ystart, values / 2)),
+            segment.colour = "gray33"
+          ),
+          direction = "y",
+          nudge_x = -.35,
+          colour = "#4e4d47",
+          size = 5,
+          inherit.aes = FALSE
+        ) +
+        annotate(
+          geom = "text",
+          x = mid_plot,
+          y = max(df$yend) * 1.1,
+          size = 16 / .pt,
+          label = paste0(
+            str_wrap(
+              paste0(
+                "Total ",
+                case_when(
+                  input$syso_level_of_detail == "All" ~ "People",
+                  input$syso_level_of_detail == "HoHsOnly" ~ "Heads of Household",
+                  TRUE ~
+                    getNameByValue(syso_level_of_detail, input$syso_level_of_detail)
+                ),
+                case_when(
+                  input$syso_hh_type == "All" ~ "",
+                  str_detect(input$syso_hh_type, "Household") == FALSE ~
```

Comment:
This will be reflected in the new sys_total_count_line function I wrote, but I think this line needs to change to:

`str_detect(getNameByValue(syso_hh_types, input$syso_hh_type), "Household") == FALSE...`

Otherwise, input$syso_hh_type might be "AC" or "AO", rather than the _name_ of that being Households with Children or Households without Children

Reply:
When I tested it, it returned the names of the values since they're named lists. When you run this in a browser, are you seeing the values instead of the names?

Reply:
Update: I can't recreate this issue, let me know if you're not getting the same as me

Reply:
Update to my update: I DID find this and corrected it like you suggested- THANKS!